pkg/schedule/schedulers/evict_slow_store.go

```diff
@@ -308,7 +350,113 @@ func (s *evictSlowStoreScheduler) IsScheduleAllowed(cluster sche.SchedulerCluste
 // Schedule implements the Scheduler interface.
 func (s *evictSlowStoreScheduler) Schedule(cluster sche.SchedulerCluster, _ bool) ([]*operator.Operator, []plan.Plan) {
 	evictSlowStoreCounter.Inc()
+	if len(s.conf.getNetworkSlowStores()) != 0 || s.conf.getMaxNetworkSlowStore() != 0 {
+		s.scheduleNetworkSlowStore(cluster)
+	}
+	return s.scheduleDiskSlowStore(cluster), nil
+}
+
+func (s *evictSlowStoreScheduler) scheduleNetworkSlowStore(cluster sche.SchedulerCluster) {
+	networkSlowStores := s.conf.getNetworkSlowStores()
+	recoveryGap := s.conf.getRecoveryDurationGap()
+
+	deleteStore := func(storeID uint64) {
+		evictedSlowStoreStatusGauge.DeleteLabelValues(s.GetName(), strconv.FormatUint(storeID, 10), string(networkSlowStore))
+		s.conf.Lock()
+		delete(s.conf.NetworkSlowStores, storeID)
+		if err := s.conf.save(); err != nil {
+			// If failed to persist, we still need to remove the store from the map.
+			log.Warn("failed to persist evict slow store config",
+				zap.Uint64("store-id", storeID),
+				zap.Error(err))
+		}
+		s.conf.Unlock()
+	}
+
+	// try to recover the network slow store if it is normal.
+	for storeID, startTime := range networkSlowStores {
+		store := cluster.GetStore(storeID)
+		if store == nil || store.IsRemoved() {
+			// Previous slow store had been removed, remove the scheduler and check
+			// slow node next time.
+			log.Info("network slow store has been removed",
+				zap.Uint64("store-id", storeID))
+
+			deleteStore(storeID)
+			continue
+		}
+
+		if calculateAvgScore(store.GetNetworkSlowScores()) <= slowStoreRecoverThreshold && uint64(time.Since(startTime).Seconds()) >= recoveryGap {
+			cluster.ResumeLeaderTransfer(storeID, constant.In)
+			deleteStore(storeID)
+		}
+	}
+
+	stores := cluster.GetStores()
+	for _, store := range stores {
+		if _, exist := networkSlowStores[store.GetID()]; exist {
+			continue
+		}
+
+		networkSlowScores := store.GetNetworkSlowScores()
+		// Remove already slow stores from the scores map to avoid duplicate detection
+		for storeID := range networkSlowStores {
+			delete(networkSlowScores, storeID)
+		}
+		if len(networkSlowScores) == 0 {
+			continue
+		}
+
+		// Check condition 1: At least one score >= networkSlowStoreIssueThreshold (95)
+		// This identifies a store with network issues to a specific peer
+		maxScoreStoreID := findMaxScoreStore(networkSlowScores, networkSlowStoreIssueThreshold)
+		if maxScoreStoreID == 0 {
+			continue
+		}
+
+		// Remove the max score store for condition 2
+		delete(networkSlowScores, maxScoreStoreID)
```

Comment:
Consider that we have removed the networkSlowScores from networkSlowScores, should we continue removing the max score?